android/core/src/main/java/com/stadiamaps/ferrostar/core/NavigationStateObserver.kt

```diff
@@ -0,0 +1,5 @@
+package com.stadiamaps.ferrostar.core
+
+interface NavigationStateObserver {
+  fun onNavigationState(state: NavigationState)
```

Comment:
I think this sounds a wee bit cleaner.

Could you also add docstrings?

```suggestion
  fun onNavigationStateUpdated(state: NavigationState)
```